### DISCUSSION TITLE
React on Changes || Data Sync to custom API

### DISCUSSION THREAD

**Discussion by dharamhbtik** [realm/realm-dotnet#3492] [Q&A]
We are using Realm's latest version for our Xamarin Forms targetting offline first approach. So we have almost 20 model class implementing IRealmObject out of which around 9 models needs to be synced back to API if there is any update.

Currently, we have created a background service that runs every 60 seconds and checks if there are any changes in those 9 models based on the flag, we select those data, convert it to JSON, and then post it to API. After a successful API post, we update the flag.

We also tried to subscribe to the realm event described at https://www.mongodb.com/docs/realm/sdk/dotnet/react-to-changes/ but there is no way to find which model changed and it keeps firing.

Looking for some solution through which I can identify the changes in those 9 models only in real-time and sync the delta to API.

**Comment by papafe** [comment]
Hi @dharamhbtik, I suppose you tried subscribing to the generic`RealmChanged` event, am I correct? If that's the case I agree that that's not enough specific for your case. 
Have you tried [registering a collection change listener](https://www.mongodb.com/docs/realm/sdk/dotnet/react-to-changes/#register-a-collection-change-listener)? This way you should get all the necessary information about changes in your collections.

**Reply by dharamhbtik** [reply]
Yes, but that does not fit in our current solution. We generally update all 9 models simultaneously and then send all models JSON in a single go. So if we subscribe each collection separately, it will be 9 subscriptions and then hard to get all changes.

**Reply by papafe** [reply]
I see. When you say you "update all 9 models" simultaneously, can you be more precise? Are you adding new elements, modifying, deleting..?

**Reply by dharamhbtik** [reply]
Like I have `List<Model1>, List<Model2>, List<Model3> `... and so on ...

 So here I am updating some properties of one or more elements in `List<Model1>` and some in List<Model2> and so on ...

**Reply by papafe** [reply]
Mmh, I don't think that there is any particular approach that you can use to get notifications on multiple classes at the same time in any case. 
I understand you have your own API, but this would be automatically done if you would use [Atlas Device Sync](https://www.mongodb.com/docs/atlas/app-services/sync/).

**Reply by dharamhbtik** [reply]
Yes, The problem is we are not using Atlas Device Sync, else there will not be any issue. Here the mechanism and architecture we are using with BackEnd API is not matching with the realm capabilities.
